pkg/logs/schedulers/ad/scheduler.go

```diff
@@ -14,38 +14,43 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
 	logsConfig "github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/schedulers"
 	"github.com/DataDog/datadog-agent/pkg/logs/service"
 	"github.com/DataDog/datadog-agent/pkg/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-var (
-	// scheduler is plugged to autodiscovery to collect integration configs
-	// and schedule log collection for different kind of inputs
-	adScheduler *Scheduler
-)
-
 // Scheduler creates and deletes new sources and services to start or stop
-// log collection on different kind of inputs.
-// A source represents a logs-config that can be defined either in a configuration file,
-// in a docker label or a pod annotation.
-// A service represents a process that is actually running on the host like a container for example.
+// log collection based on information from autodiscovery.
+//
+// This type implements  pkg/logs/schedulers.Scheduler.
 type Scheduler struct {
-	sources  *logsConfig.LogSources
-	services *service.Services
+	mgr                schedulers.SourceManager
+	listener           *adListener
+	sourcesByServiceID map[string]*logsConfig.LogSource
```

Comment:
Note that sources are not unique by service ID; see https://github.com/DataDog/datadog-agent/pull/12185.